File: demo/common/src/main/java/com/serious/demo/TestRemoteComponent.java
package com.serious.demo;
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

import com.serious.service.Component;
import com.serious.service.ComponentInterface;
import org.springframework.web.bind.annotation.RestController;

/**
 * @author Andreas Ernst
 */
@ComponentInterface(
        services = {
                TestRemoteRestService.class
        })
@RestController
public interface TestRemoteComponent extends Component {
}

File: demo/common/src/main/java/com/serious/demo/TestRemoteComponent.kt
package com.serious.demo
/*
* @COPYRIGHT (C) 2023 Andreas Ernst
*
* All rights reserved
*/

import com.serious.service.Component
import com.serious.service.ComponentInterface
import org.springframework.web.bind.annotation.RestController

@ComponentInterface(name = "TestRemoteComponent", services = [TestRemoteRestService::class])
@RestController
interface TestRemoteComponent : Component
